web/crux-ui/src/validations/container.ts

```diff
@@ -186,16 +188,41 @@ const storageRule = yup
   .nullable()
   .optional()
 
-const portConfigRule = yup
-  .array(
-    yup.object().shape({
-      internal: portNumberRule,
-      external: portNumberOptionalRule,
-    }),
+const createOverlapTest = (
+  schema: yup.NumberSchema<number, AnyObject, number>,
+  portRanges: ContainerConfigPortRange[],
+  field: Exclude<keyof ContainerConfigPortRange, 'id'>,
+) =>
+  // eslint-disable-next-line no-template-curly-in-string
+  schema.test('port-range-overlap', '${path} overlaps port ranges', (value, _) =>
+    portRanges.length > 0 ? !portRanges.some(it => value >= it[field].from && value <= it[field].to) : true,
```

Comment:
Please make a body for the lambda and declare some const to make this more readable.